client/js/ace/mode-fbp.js

```diff
@@ -54,15 +54,14 @@ var ShHighlightRules = function() {
     var pointFloat = "(?:(?:" + intPart + "?" + fraction + ")|(?:" + intPart + "\\.))";
     var exponentFloat = "(?:(?:" + pointFloat + "|" +  intPart + ")" + ")";
     var floatNumber = "(?:" + exponentFloat + "|" + pointFloat + ")";
-    var fileDescriptor = "(?:&" + intPart + ")";
-    var fileDescriptor = "[a-zA-Z0-9_-]+\\/[a-zA-Z0-9_-]+";
+    var nodeTypes = "[(]([^:)]+)[:)]";
 
-    var variableName = "[a-zA-Z_][a-zA-Z0-9_]*";
-    var variable = "(?:(?:\\$" + variableName + ")|(?:" + variableName + "=))";
+    var nodeOptions = "[a-z_][a-z0-9_]*";
+    var variable = "(?:(?:\\$" + nodeOptions + ")|(?:" + nodeOptions + "=))";
 
     var builtinVariable = "(?:\\$(?:SHLVL|\\$|\\!|\\?))";
 
-    var func = "[A-Z0-9]+";
+    var ports = " [A-Z0-9\\[\\]]+ ";
```

Comment:
like above, no reason why port names cannot be lowercase, just that we do not use them. I'd not forbid this in the highlight.

also, to be more exact you can force only number after `[`, and require a `]` to have port arrays, something like:
```js
var ports = "[a-zA-Z_][a-zA-Z0-9_]*(|\\[[0-9]+\\])";
```

I'm also not sure why a leading space is required.